test/FastRouteRouterTest.php

```diff
@@ -823,4 +824,20 @@ public function testGetAllAllowedMethods()
             $result->getAllowedMethods()
         );
     }
+
+    public function testUserDefinedDispatcherCallbackGroupPosBased()
+    {
+        $route1 = new Route('/foo', $this->getMiddleware());
+
+        $router = new FastRouteRouter(null, function ($data) {
+            return new GroupPosBased($data);
```

Comment:
Hm, maybe this test should be even more generic, so we should prophesize here Dispatcher instance ? What do you think?

Reply:
With last commit we prophesize the Dispatcher Interface instead the `GroupCountBased` in the `getRouter` method of the unit test. 

Therefor I think this test isn't needed anymore, we could test if the dispatcherCallback is invoked correctly.

Reply:
@SanderSander sounds reasonable; can you push a patch with that change to your branch? We can then do final review and merge.

Thanks!

Reply:
Done, @weierophinney I Improved the test custom implementations of the Dispatcher interface